### PR TITLE
chagned behavior of com.gs.return with backward compatibility with v1

### PR DESCRIPTION
**Pull request by mastersilv3r**
Related to #980

### CODE REVIEW

File: src/functions/com/gs/return.ts [hunk 1]
```diff
@@ -6,20 +6,28 @@ import { PlainObject } from "../../../types";
...
+  const v1Compatible = ctx.config.returnV1Compatible;
+
+  if (v1Compatible) {
+    return { success: true, code: 200, data: args, exitWithStatus: true };
   } else {
-    success = v1Compatible ? true : (success !== undefined && success !== null ? success : true);
-    code = v1Compatible ? 200 : code || 200;
+    success = args.hasOwnProperty('success') ? args.success : true;
+    code = args.code || 200;
+    data = args.hasOwnProperty('data') ? args.data : args;
```
Comment (gurjotmindgrep) [inline src/functions/com/gs/return.ts:29]: What will happen in this scenario?
```
fn: com.gs.return
args:
  code: 400
  success: false
  message: "Bad request"
```

I guess we would get the following response.
```
{
  "data": {
    "data": {
      "code": 400,
      "success": false,
      "message": "Bad request"
    }
  }
}
```

I think we should remove success and code from args.

Comment (mastersilv3r) [reply src/functions/com/gs/return.ts:29]: Makes sense.. can you please do it and merge.

Comment (mastersilv3r) [reply src/functions/com/gs/return.ts:29]: As well, update the issue documentation..clearly stating this..

Comment (gurjotmindgrep) [reply src/functions/com/gs/return.ts:27]: Done

**Review by mastersilv3r — COMMENTED** [review]
Shouldn't we make this change in transform.ts and use that in return.ts? Please make sure the forAuth and authz logic is not broken. Task level and eventsource level authz should work with both return and transform the same way

**Pull request closed without merging**